pkgs/tools/networking/guardian-agent/default.nix

```diff
@@ -0,0 +1,42 @@
+{ buildGoPackage
+, fetchFromGitHub
+, lib
+, openssh
+, autossh
+, x11_ssh_askpass
+}:
+
+buildGoPackage rec {
+  pname = "guardian-agent";
+  version = "0.7.2-beta";
+
+  src = fetchFromGitHub {
+    owner = "StanfordSNR";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "05269y944hcabn6dqa66387rdhx81vcqcyjv6m1hdbz5ba8j7mqn";
+  };
+
+  goPackagePath = "github.com/StanfordSNR/guardian-agent";
+
+  goDeps = ./deps.nix;
+
+  nativeBuildInputs = [ openssh autossh x11_ssh_askpass ];
```

Comment:
`autossh` doesn't seem to be needed for the build.

Instead it's a runtime dependency of this script
* https://github.com/StanfordSNR/guardian-agent/blob/bc8088c725391a62c25edb625ea5159474de2041/scripts/sga-guard.

Judging from how it works it would be best to make a wrapper script with `wrapProgram` that adds the appropriate
entries to `PATH`.

```nix
postFixup = ''
  wrapPrgoram $out/bin/sga-guard \
     --prefix PATH : "$out/bin" \
     --prefix PATH : "${autossh}/bin"
'';
```

##### [`wrapProgram` described](https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/setup-hooks/make-wrapper.sh#L138)

Reply:
As for `openssh` and `x11_ssh_askpass`, does the go code in `guardian-agent` need their executables at runtime?

Reply:
All of these are runtime dependecies, moved them to ```propagatedBuildInputs```, the ```wrapProgram``` doesn't work with a script, I'm not sure if there is an alternative?

Reply:
I thought about ``wrapProgram``ing the sga-guard-bin, however the script execs it with some specific parameters.

Reply:
I ended up patching the script to pull ``autossh``.